keyboards/preonic/keymaps/jpe230/keymap.c

```diff
@@ -0,0 +1,116 @@
+  ),
+
+  [2] = LAYOUT_preonic_grid(
+  //,-----------------------------------------------------------------------------------------------------------.
+       KC_GRV,  KC_F11,  KC_F12,   KC_F3,   KC_F4,   KC_F5,   KC_F6,   KC_F7,   KC_F8,   KC_F9,  KC_F10, KC_BSPC,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+       KC_ESC, KC_EXLM,   KC_AT, KC_HASH,  KC_DLR, KC_PERC, KC_CIRC, KC_AMPR, KC_ASTR, KC_LPRN, KC_RPRN, KC_BSPC,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      KC_CAPS, _______, KC_MS_U, _______, _______, _______, KC_MINS,  KC_EQL, KC_LBRC, KC_RBRC, KC_BSLS,  KC_GRV,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      KC_LSFT, KC_MS_L, KC_MS_D, KC_MS_R, _______, _______, KC_UNDS, KC_PLUS, KC_LCBR, KC_RCBR, KC_PIPE, KC_TILD,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      _______, _______, _______,  MO(3),  _______, KC_BTN1, _______, _______, _______, _______, _______, _______
+  //`-----------------------------------------------------------------------------------------------------------'
+  ),
+
+  [3] = LAYOUT_preonic_grid(
+  //,-----------------------------------------------------------------------------------------------------------.
+       KC_ESC,    KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,  KC_DEL,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      KC_MUTE, KC_WBAK, KC_WFWD,   KC_F7,   KC_F8,   KC_F9, _______,    KC_7,    KC_8,    KC_9, _______, QK_BOOT,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      KC_VOLU, _______, KC_MNXT,   KC_F6,   KC_F5,   KC_F6, _______,    KC_6,    KC_5,    KC_4, _______, QK_MAKE,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+      KC_VOLD, KC_MPRV, KC_MNXT,   KC_F1,   KC_F2,   KC_F3, KC_LSRS,    KC_1,    KC_2,    KC_3, _______, DB_TOGG,
+  //|--------+--------+--------+--------+--------+--------|--------+--------+--------+--------+--------+--------|
+       MU_TOG,  MU_MOD, _______, _______, _______, KC_LSRE, _______, KC_LERS, _______, _______, _______,  EE_CLR
+  //`-----------------------------------------------------------------------------------------------------------'
+  )
+};
+
+/* 
+Logic to shift between the bars:
+    KC_LSRE => Left Bar - Space... Right Bar - Enter
+    KC_LERS => Left Bar - Enter... Right Bar - Space
+    KC_LSRS => Left Bar - Space... Right Bar - Space
+
+    Hack into dynamic_keymap_set_keycode and change the value of the keymap.
```

Comment:
If you're not using VIA, you could just use `DYNAMIC_KEYMAP_ENABLE=yes` in your rules.mk, instead, in theory.

Reply:
I do use VIA, but it is faster to have a way to swap those keycodes inside the kb, specially when alt-tabbing from a game and having to write something :p.

Resolving the issue since it is the intended behavior :)